zebrad/tests/common/shielded_scan/scans_for_new_key.rs

```diff
@@ -105,7 +105,12 @@ pub(crate) async fn run() -> Result<()> {
 
     tracing::info!("started scan task, sending register keys message with zecpages key to start scanning for a new key",);
 
-    scan_task.register_keys(parsed_keys)?;
+    scan_task.register_keys(
+        parsed_keys
+            .into_iter()
+            .map(|key| (key.0, Some(key.1 .2 .0)))
```

Comment:
Optional/Nitpick/Subjective: this may be slightly more clear if de-structured.

```suggestion
            .map(|(key, (_, _, Height(h)))| (key, Some(h)))
```

Reply:
I disagree. It's way more clear.